lib/improver/weights.py

```diff
@@ -548,16 +547,13 @@ def process(self, cube, coord_name):
         # Rescale width and midpoint if in different units to the coordinate
         if coord_units != self.parameters_units:
             self.width = self.parameters_units.convert(self.width, coord_units)
-            self.midpoint = self.parameters_units.convert(self.midpoint,
-                                                          coord_units)
 
-        weights = self.triangular_weights(coord_vals)
+        weights = self.triangular_weights(coord_vals, midpoint)
 
         return weights
 
     def __repr__(self):
         """Represent the configured plugin instance as a string."""
-        string = ("<ChooseDefaultTriangularWeights width={:4.1f},"
-                  " midpoint={:4.1f}>")
-        desc = string.format(self.width, self.midpoint)
+        string = "<ChooseDefaultTriangularWeights width={:4.1f}>"
```

Comment:
I think that it would be better to name this variable `msg` as [string](https://docs.python.org/2/library/string.html) is actually an in-built module.